src/pages/Main/Asides/ImageSlider.js

```diff
@@ -0,0 +1,187 @@
+            ...newState,
+            imageCounter: newImageCounter,
+            animation: { transform, transition },
+          };
+        });
+      }, animationTime);
+    }
+
+    if (imageCounter === imageList.length) {
+      setTimeout(() => {
+        this.setState(prevState => {
+          const newState = { ...prevState };
+
+          let xLocation = -width;
+          let transform = `translateX(${xLocation}px)`;
+          let newImageCounter = 0;
+
+          return {
+            ...newState,
+            imageCounter: newImageCounter,
+            animation: { transform, transition },
+          };
+        });
+      }, animationTime);
+    }
+  }
+
+  render() {
+    const { setImageCounter } = this;
+    const { animation, imageCounter } = this.state;
+    const { imageList, imgSize, description } = this.props;
+
+    const { buttonRender, buttonWrapClassName, buttonClassName, buttonText } =
+      this.props;
+
+    const imageDescription = description ? (
+      <div className="imgDescription">
+        <span className="imgFlag">신메뉴 오픈</span>
+        <span className="imgDescTitle">
+          한 여름의 힐링캠핑 어쩌구 저저구 꾸깃 어쩌
+        </span>
+        <span className="imgDescText">꾸깃이 발행하는 웹 매거진</span>
+      </div>
+    ) : null;
+
+    const _imageList = imageList.map((image, idx) => {
+      const { name, url } = image;
+      return (
+        <div
+          key={idx + 1}
+          className={'imgWrap'}
+          aria-hidden={idx + 1 === imageCounter ? 'false' : 'true'}
+        >
+          <img
+            alt={name}
+            src={url}
+            width={imgSize.width}
+            height={imgSize.height}
+          />
+          {imageDescription}
+        </div>
+      );
+    });
+
+    return (
+      <div className="imgSliderWrap">
+        <div className="imgSlider" style={animation}>
+          <div
+            key={-1}
+            className="imgWrap"
+            aria-hidden={imageCounter === -1 ? 'false' : 'true'}
+          >
+            <img
+              alt={imageList[imageList.length - 1].name}
+              src={imageList[imageList.length - 1].url}
+              width={imgSize.width}
+              height={imgSize.height}
+            />
+          </div>
+          {_imageList}
+          <div
+            key={imageList.length}
+            className="imgWrap"
+            aria-hidden={imageCounter === imageList.length ? 'false' : 'true'}
+          >
+            <img
+              alt={imageList[0].name}
+              src={imageList[0].url}
+              width={imgSize.width}
+              height={imgSize.height}
+            />
+          </div>
+        </div>
+        {buttonRender ? (
+          <ImageSliderButton
+            buttonRender={buttonRender}
+            buttonWrapClassName={buttonWrapClassName}
+            buttonClassName={buttonClassName}
+            buttonText={buttonText}
+            handleClick={e => {
+              setImageCounter(e);
+            }}
```

Comment:
```suggestion
            handleClick={setImageCounter}
```
굳이 인라인 함수선언해서 사용할 필요 없어보입니다!

Reply:
넵 알겠습니다.